### PR TITLE
test: lambda template testing

### PR DESCRIPTION
**Pull request by Victor-C-Zhang**

### CODE REVIEW

**Review by fahadmohammed01 — APPROVED** [review]

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -28,13 +28,17 @@ jobs:
         continue-on-error: true
       - name: Build with Maven (Windows)
         env:
-          AWS_REGION: us-west-2
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (MikeDombo) [inline .github/workflows/maven.yml:31]: For GG we cannot have this. Our tests which run in github may not access AWS resources.

Fine for you, for now.

File: src/main/java/com/aws/greengrass/lambdatransformer/common/Constants.java [hunk 2]
```diff
@@ -59,7 +59,7 @@
 
     public static final int LAMBDA_RECIPE_DEFAULT_MAX_INSTANCE_COUNT = 100;
 
-    public static final int LAMBDA_RECIPE_DEFAULT_LAMBDA_MEMORY_IN_KILO_BYTES = 16_384; // 16 MB
+    public static final int LAMBDA_RECIPE_DEFAULT_LAMBDA_MEMORY_IN_KILO_BYTES = 16_000;
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/lambdatransformer/common/Constants.java:62]: why change it?

File: src/main/java/com/aws/greengrass/lambdatransformer/common/models/LambdaEventSource.java [hunk 3]
```diff
@@ -20,9 +19,7 @@
     private LambdaEventSourceType type;
 
     private enum LambdaEventSourceType {
-        @JsonProperty("pubsub")
-        PUBSUB,
-        @JsonProperty("iotcore")
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/lambdatransformer/common/models/LambdaEventSource.java:25]: why did you remove the property names? "pubsub" and "iotcore" is correct isn't it?

File: src/main/java/com/aws/greengrass/lambdatransformer/common/Constants.java (hunk 2 above)
Comment (Victor-C-Zhang) [reply src/main/java/com/aws/greengrass/lambdatransformer/common/Constants.java:62]: to match cloud behavior

File: src/main/java/com/aws/greengrass/lambdatransformer/common/models/LambdaEventSource.java (hunk 3 above)
Comment (Victor-C-Zhang) [reply src/main/java/com/aws/greengrass/lambdatransformer/common/models/LambdaEventSource.java:25]: no, they are PUB_SUB and IOT_CORE, respectively

**Review by MikeDombo — APPROVED** [review]
